core/src/test/java/bio/terra/pearl/core/service/kit/KitRequestServiceTest.java

```diff
@@ -31,19 +40,113 @@ public void testRequestKit() throws Exception {
         profile.setGivenName("Alex");
         profile.setFamilyName("Tester");
         profile.setPhoneNumber("111-222-3333");
-        profileService.update(profile);
-        var expectedSentToAddress = "{\"firstName\":\"Alex\",\"lastName\":\"Tester\",\"street1\":null,\"street2\":null,\"city\":null,\"state\":null,\"postalCode\":null,\"country\":null,\"phoneNumber\":\"111-222-3333\"}";
+        profile.getMailingAddress().setStreet1("123 Fake Street");
+        profileService.updateWithMailingAddress(profile);
+        var expectedSentToAddress = PepperKitAddress.builder()
+                .firstName("Alex")
+                .lastName("Tester")
+                .street1("123 Fake Street")
+                .phoneNumber("111-222-3333")
+                .build();
 
         var sampleKit = kitRequestService.requestKit(adminUser, enrollee, "testRequestKit");
 
         Mockito.verify(mockPepperDSMClient)
                 .sendKitRequest(eq(enrollee), any(KitRequest.class), any(PepperKitAddress.class));
         assertThat(sampleKit.getCreatingAdminUserId(), equalTo(adminUser.getId()));
         assertThat(sampleKit.getEnrolleeId(), equalTo(enrollee.getId()));
-        assertThat(sampleKit.getSentToAddress(), equalTo(expectedSentToAddress));
+        assertThat(objectMapper.readValue(sampleKit.getSentToAddress(), PepperKitAddress.class),
+                equalTo(expectedSentToAddress));
         assertThat(sampleKit.getStatus(), equalTo(KitRequestStatus.CREATED));
     }
 
+    @Transactional
+    @Test
+    public void testUpdateKitStatus() throws Exception {
+        var adminUser = adminUserFactory.buildPersisted("testUpdateKitStatus");
+        var enrollee = enrolleeFactory.buildPersisted("testUpdateKitStatus");
+        var kitType = kitTypeFactory.buildPersisted("testUpdateKitStatus");
+        var kitRequest = kitRequestFactory.buildPersisted("testUpdateKitStatus",
+                adminUser.getId(), enrollee.getId(), kitType.getId());
+
+        var response = PepperDSMKitStatus.builder()
+                .kitId(kitRequest.getId().toString())
+                .currentStatus("SENT")
+                .build();
+        when(mockPepperDSMClient.fetchKitStatus(kitRequest.getId())).thenReturn(response);
+
+        var sampleKitStatus = kitRequestService.updateKitStatus(kitRequest.getId());
+
+        assertThat(sampleKitStatus.getCurrentStatus(), equalTo("SENT"));
+    }
+
+    @Transactional
+    @Test
+    public void testUpdateAllKitStatuses() throws Exception {
```

Comment:
There's a lot to arrange for this test. I don't love how long this test function is, but part of what it's verifying is that it updates each kit status individually, so maybe it's necessary complexity. I'm open to opinions.